src/control/cmd/daos/container.go

```diff
@@ -562,14 +589,33 @@ func (cmd *containerListCmd) Execute(_ []string) error {
 			"unable to list containers for pool %s", cmd.PoolID())
 	}
 
+	contInfo := make([]*containerInfo, 0, len(contIDs))
+	for _, contID := range contIDs {
+		if cmd.Verbose {
+			info, err := cmd.containerInfo(&contID.UUID)
```

Comment:
In a production setting where a pool will have containers owned by different users, isn't this going to fail everytime with ENOPERM for containers they don't have access to?
so i question the usefulness in this case.

Reply:
fair enough, I'm not using multiple owners at the moment.
On the other hand, the ones without access can be omitted, or perhaps reading container attributes allowed for anyone ?

Reply:
> On the other hand, the ones without access can be omitted

But that is not really a container list anymore (and not even a verbose list), is it? it would very weird that list-containers return one thing, and list-containers --verbose returns a subset. sounds like you need a completely different option to return containers that you are an owner (or have read permissions on) and return those.  

> perhaps reading container attributes allowed for anyone

the problem is that attributes you are reading here for POSIX containers are not really attributes from the container perspective, but just regular data (POSIX containers store them in the SB object which is just a regular object from container permissions sake) so it involves a regular object fetch that you should have read permissions for.

Reply:
@kccain pointed out yesterday that there is a new-ish container API that might be useful here: https://github.com/daos-stack/daos/blob/11e2bddb4575df47092a91ec9cc23973a0b5d067/src/include/daos_pool.h#L611

In theory, that should allow us to construct a filter to only show containers meeting certain criteria. I'm not sure if there is a way to express read permission as a filter criterion, though.

Reply:
Not related to container owners -  but applicable to outputting some common properties: This new interface combines list containers behavior with the basic part of a container query (the daos_cont_info_t information).  However it does not return properties. I tried to leave a placeholder in one of the data structures in case we might decide to try to add properties in later, but didn't think too exactly about how it would be implemented. Except to think that perhaps a more sophisticated implementation of the cart bulk transfer could be done -  i.e., list_cont_bulk_create() that in this implementation currently refers to a single linear buffer of size ncont * sizeof(struct daos_pool_cont_info2). Maybe the pci_reserved[] pointers in that structure could have some additional dynamically allocated storage to accommodate properties coming back from the engine in the bulk transfer.

Reply:
Looks like permission issue might get on the way with the verbose output.
I do agree with @mchaarawi that skipping containers without proper access does not express `list` command semantic at all.

I guess tooling have to deal with containers one by one 😭 
Maybe one day with another interface it would be straightforward to implement 😬